src/lib/GClient.ts

```diff
@@ -55,6 +136,35 @@ export class GClient<Ready extends boolean = boolean> extends Client<Ready> {
 	}
 
 	// eslint-disable-next-line @typescript-eslint/no-unused-vars
+	/**
+	 * The method that returns original types from {@link GClientOptions.database}
+	 * @param {any} _ Prototype of your class to get types.
+	 * @returns {Database}
+	 * @example TypeScript example
+	 * ```typescript
+	 * import { MongoDBProvider } from 'gcommands/dist/providers/MongoDBProvider';
+	 *
+	 * const client = new GClient({
+	 * 		..settings,
+	 * 		database: new MongoDBProvider('mongodb://localhost:27017/database')
+	 * })
+	 *
+	 * const db = client.getDatabase(MongoDBProvider.prototype);
+	 * // Types working in `db`
+	 * ```
+	 * @example JavaScript example
+	 * ```javascript
+	 * const { MongoDBProvider } = require('gcommands/dist/providers/MongoDBProvider');
+	 *
+	 * const client = new GClient({
+	 * 		..settings,
+	 * 		database: new MongoDBProvider('mongodb://localhost:27017/database')
+	 * })
+	 *
+	 * const db = client.getDatabase(MongoDBProvider.prototype);
+	 * // Types working in `db`
+	 * ```
+	 */
 	public getDatabase<Database>(_?: Database): Database {
```

Comment:
```suggestion
	/**
	 * The method that returns the database provided in {@link GClientOptions}
	 * @returns {Database}
	 * @example TypeScript example
	 * ```typescript
	 * import { MongoDBProvider } from 'gcommands/dist/providers/MongoDBProvider';
	 *
	 * const client = new GClient({
	 * 		..settings,
	 * 		database: new MongoDBProvider('mongodb://localhost:27017/database')
	 * })
	 *
	 * const db = client.getDatabase();
	 * // returns <MongoDBProvider>
	 * ```
	 * @example JavaScript example
	 * ```javascript
	 * const { MongoDBProvider } = require('gcommands/dist/providers/MongoDBProvider');
	 *
	 * const client = new GClient({
	 * 		..settings,
	 * 		database: new MongoDBProvider('mongodb://localhost:27017/database')
	 * })
	 *
	 * const db = client.getDatabase();
	 * // returns <MongoDBProvider>
	 * ```
	 */
	public getDatabase<Database>(): Database {
```